lib/spell.js

```diff
@@ -547,10 +530,12 @@ class Spell {
       hints.push(Hint.build(opts.hint));
```

Comment:
added:

```js
Post.find(conditions, {
  hint: {
     index: 'idx_title',
     type: INDEX_HINT_TYPE.force,
     scope: INDEX_HINT_SCOPE.groupBy,
  },
});
```

Reply:
that will throw ` Unknown optimizer hint` error, because of `new Hint(Object)` in `Spell`'s constructor, `Hint` isn't polymorphic.

Reply:
nope, options.hint is dispatched with `Hint.build()`, which handles objects as well. If `options.hint` is an object and has `options.hint.index`, it will get instantiated with IndexHint. see the reverted hint.test.js